design-docs/opmodes-commandbased.md

```diff
@@ -0,0 +1,205 @@
+# Summary
+
+This document describes how opmodes are implemented in command-based projects.
+
+# Motivation
+
+See [opmodes](opmodes.md) for the motivation for adding operator selectable opmodes to the core robot structure.  For command-based programs, the entire robot program is structured around command-based subsystems and commands tied to robot states and inputs, with everything constructed during initialization, so for more natural integration with the rest of the command-based framework, a different approach than the annotation-based approach used for periodic and linear opmodes is warranted.
```

Comment:
Can you explain more about the problem caused by attempting to use an annotation-based approach in a command-based project? I'm not sure I understand why this wouldn't work and feel fairly natural. Is there some compile-time vs runtime difference I'm missing here?

Reply:
Using annotations in what way?  It wouldn't make sense to use them on classes like for the non-command-based modes, because all command-based things are set up during robot construction.  Maybe function annotations?  The reason we don't allow a mix of non-command and command-based opmodes is because of the assumptions surrounding subsystem periodic and the scheduler always being called (see @jwbonner's earlier comment).